scipy/optimize/optimize.py

```diff
@@ -2102,7 +2105,10 @@ def _minimize_scalar_golden(func, brack=None, args=(),
     f1 = func(*((x1,) + args))
     f2 = func(*((x2,) + args))
     funcalls += 2
-    while (numpy.abs(x3 - x0) > tol * (numpy.abs(x1) + numpy.abs(x2))):
+    nit = 0
+    for i in xrange(maxiter):
```

Comment:
Yurii Shevchuk a écrit :
> For instance it doesn't work when |maxiter = 1|. In this case |nit = 0|.
>

Then, wouldn't `for nit in xrange(1, maxiter + 1):` work?